File: PHP/49.group-anagrams.php
<?php
class Solution {

    /**
     * @param String[] $strs
     * @return String[][]
     */
    function groupAnagrams($strs) {
        if (empty($strs)) {
            return [];
        }
        $map = [];
        foreach ($strs as $str) {
            $arr = [];
            for ($i=0; $i < strlen($str); $i++) {
                $arr[] = $str[$i];
            }
            sort($arr);
            $key = implode("", $arr);
            $map[$key][] = $str;
        }
        return array_values($map);
    }
}

File: Python/49.group-anagrams.py
'''
1. 利用排序的字符串作为键值
'''
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        m = {}
        for sub_str in strs:
            key = "".join((lambda x:(x.sort(), x)[1])(list(sub_str)))
            if key not in m:
                m[key] = []
            m[key] += [sub_str]
        return list(m.values())
